src/prank_serial.cpp

```diff
@@ -0,0 +1,140 @@
+    ifstream cin(input_file);
+
+    int id, out;
+    while(cin>>id) {
+        cin >> out;
+        if(!pages.count(id)) {
+            pages[id] = Page();
+            num_pages++;
+        }
+        if(!pages.count(out)) {
+            pages[out] = Page();
+            num_pages++;
+        }
+        pages[out].indegree++;
+        pages[out].in_ids.push_back(id);
+        pages[id].outdegree++;
+    }
+
+
+    cout << "Load time: " << (double)(clock() - tStart)/CLOCKS_PER_SEC <<endl;
+    //cout<<"Loaded "<<num_pages<<" pages"<<endl;
+
+
+    // initialize pageRank
+    float equal_prob = 1.0 / num_pages;
+    for(auto& x : pages)
+        x.second.score = equal_prob;
+
+    float broadcastScore;
+
+    //omp_set_num_threads(num_threads);
+    //PHASE 2 : Page ID Distribution
+    //TODO: ID distribution logic for different threads comes here.
+
+
+    //PHASE 3 : iteratively update ranks
+    //Once page distribution is done, we can spawn the threads
+    //The code below simply distributes the pages across threads based on page id, we need to change it
+    clock_t aStart = clock();
+    for(int j=0;j<num_iters;j++){
+        clock_t tStart = clock();
+        broadcastScore = 0.0;
+
+        // evaluate the score for each node
+        for(int i=0;i<pages.size();i++){
+
+            // find the ID corresponding to the i-page
+            int idx = i;
+
+            // used to store the new score
+            pages[idx].score_new = 0.0;
+
+            // if the node has no outgoing edges, then add its value to the broadcast score
+            if(!pages[idx].outdegree)
+                broadcastScore += pages[idx].score;
+
+            // iterate over all the vertices with an incoming edge to compute the new value of the vertices
+            if(pages[idx].indegree>0){
+                for(vector <int> :: iterator it = pages[idx].in_ids.begin(); it != pages[idx].in_ids.end(); ++it)
+                    pages[idx].score_new += pages[*it].score / pages[*it].outdegree;
+            }
+
+            // apply pageRank equation
+            pages[idx].score_new = alpha * pages[idx].score_new + (1.0 - alpha) / num_pages;
+        }
+
+        // update broadcast score
+        broadcastScore = alpha * broadcastScore / num_pages;
+        for(auto& x : pages){
+
+            // add the global broadcast score to each edge to compute its final score
+            x.second.score = x.second.score_new + broadcastScore;
+        }
+        //cout<<"Iteration "<<j<<" completed in "<<(double)(clock() - tStart)/CLOCKS_PER_SEC<<"s"<<endl;
+    }
+
+    //Print input file name followed by time taken to converge
+    cout<< "Iteration time: " << (double)(clock() - aStart)/CLOCKS_PER_SEC <<endl;
+
+    //PHASE 4: Write the result
+
+    string result_path = input_file + string("_serial_ranks") ;
+    ofstream result_file(result_path);
+    for(auto& x : pages){
+        result_file << x.first << " " << x.second.score << "\n";
+    }
+    result_file.close();
```

Comment:
Probably a brain fart on my part, but I cannot seem to find the resulting file from this when running:
`./prank_serial 9, 100, 0.850000, road_usa`

Reply:
is it printing rest of the output to console ? like time taken and stuff ?

Reply:
I get:
```5.18571%
``` The % is highlighted black (might mean some special?)

Reply:
This shouldnt be happening, lemme know if you can get on a call for this if you are free.. should be quick

Reply:
I can. One more quick data point: I must have messed something up when merging into eval. When I build on develop I get the following:
```
Loading graph data...
Graph data loaded in 5.2142s
Loaded 4682830 pages
```

Although I still don't see the file.